### ISSUE TITLE
Don't early stop, just save best

### ISSUE THREAD

**Issue by schrum2** [schrum2/MarioDiffusion#84]
train_diffusion now uses early stopping by default, but instead I would like to simply save the best model during training and save that at the end. So, the loss and caption score will still be monitored, but will never cause training to stop. At the end of training, the final model saved should be the one that had either the lowest validation loss or highest caption score (add a parameter for this). Basically, this means that the official result of training the model may not be the model from the final epoch.
